File: MainGame/FrontEndToolFS/SebastianPlanet/NoiseSettings.fs
namespace FrontEndToolFS.SebastianPlanet

open Godot

type FilterType =
    | Simple = 0
    | Rigid = 1

type NoiseSettings() =
    member val filterType = FilterType.Simple with get, set
    member val strength = 1f with get, set
    // 1 ~ 8
    member val numLayers = 1 with get, set
    member val baseRoughness = 1f with get, set
    // 1.0 ~ 10.0
    member val roughness = 2f with get, set
    // 0.0 ~ 1.0
    member val persistence = 0.5f with get, set
    member val center = Vector3.Zero with get, set
    member val minValue = 0f with get, set
    member val weightMultiplier = 0.8f with get, set

File: MainGame/game/SebastianPlanet/Settings/NoiseSettings.cs
using FrontEndToolFS.SebastianPlanet;
using Godot;

namespace ZeromaXPlayground.game.SebastianPlanet.Settings;

[Tool]
[GlobalClass]
public partial class NoiseSettings : Resource, INoiseSettings
{
    [Export] public FilterType FilterType { get; set; } = FilterType.Simple;
    [Export] public float Strength { get; set; } = 1f;
    [Export(PropertyHint.Range, "1, 8")] public int NumLayers { get; set; } = 1;
    [Export] public float BaseRoughness { get; set; } = 1f;

    [Export(PropertyHint.Range, "1.0, 10.0")]
    public float Roughness { get; set; } = 2f;

    [Export(PropertyHint.Range, "0.0, 1.0")]
    public float Persistence { get; set; } = 0.5f;

    [Export] public Vector3 Center { get; set; } = Vector3.Zero;
    [Export] public float MinValue { get; set; }
    [Export] public float WeightMultiplier { get; set; } = 0.8f;
}